### PR TITLE
Gemify teacup

### PR DESCRIPTION
**Pull request by ConradIrwin**
It's nice to be able to use teacup as a gem :). Unfortunately the technique for doing this is somewhat hacky, due to lack of real support from RubyMotion.

The technique I use is based on http://reality.hk/posts/2012/05/22/create-gems-for-rubymotion/, but made a bit more robust for users who are customizing app.files directly.

### CODE REVIEW

File: lib/teacup.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#
+class << Motion::Project::App
+
+  def setup_with_teacup
+    setup_without_teacup do |app|
+      yield app
+
+      dirs = %w(teacup teacup/core_extensions)
+      files = dirs.map{ |dir| Dir.glob("#{File.dirname(__FILE__)}/#{dir}/*.rb") }.flatten
+      app.files = files + (app.files - files)
```
Comment (farcaller) [inline lib/teacup.rb:35]:
``` ruby
app.files = files | app.files
```

Though it's a mistake to include teacup files directly, so it's safe to assume no duplicates

**Comment by farcaller** [comment]
Okay, now I see the point. But why do you redefine the #setup method?

I see two other options (not counting the app.gem support):
1. Make it clean to the user that he must use `app.files += [ ... ]`, that is, preserve the app.files content.
2. Override the #files in config and put the magic down there.

I really love the method # 1, it's most clean and unobtrusive.

**Comment by ConradIrwin** [comment]
If the user wants to require stuff _before_ his app directory, then 1. doesn't work. See https://github.com/ConradIrwin/Commune/blob/master/Rakefile. 2. Felt even nastier than this approach to me.

**Comment by ConradIrwin** [comment]
This stuff conflicts badly with the tests (poor committing from me). I've pushed a merge at https://github.com/ConradIrwin/teacup/tree/merge-tests-gemification

**Comment by farcaller** [comment]
I suggest to close this for now, as we might see a ruby motion native support for gems in a future release or two. Meanwhile we can focus on the other stuff.

**Pull request closed without merging**